### PR TITLE
Implement `getrevocationtxs`

### PR DESCRIPTION
**Pull request by darosior**
This implements the `getrevocationtxs` RPC, as well as a few database interface stuff and fixes the documentation (~~but not yet the implem!~~ now implemented) for the vault identifier.

~~I'll add some tests tomorrow but you can already give it a look, i tested it manually :tm:~~ Added a functional test

### CODE REVIEW

**Comment by darosior** [comment]
So the next suspect will be https://github.com/re-vault/revaultd/blob/master/doc/API.md#getonchaintxs, which will need the db interface anyways so dead code won't last :)

File: src/daemon/database/interface.rs [hunk 1]
```diff
@@ -184,65 +192,197 @@ pub struct DbVault {
     pub derivation_index: ChildNumber,
 }
 
+fn vault_from_row(row: &Row) -> rusqlite::Result<DbVault> {
```
Comment (edouardparis) [inline src/daemon/database/interface.rs:195]: May I suggest:

```rust
impl TryFrom<&Row<'_>> for DbVault {
    type Error = rusqlite::Error;
    fn try_from(row: &Row) -> Result<Self, Self::Error> {
```
then 
```diff
-         vault_from_row,
+        |row: &Row| row.try_into(),
```

File: doc/API.md [hunk 2]
```diff
@@ -92,14 +92,14 @@ Note that the `scriptPubKey` is implicitly known as we have the vault output Min
...
 
 #### Request
 
-| Parameter | Type         | Description                                                                                     |
-| --------- | ------------ | ----------------------------------------------------------------------------------------------- |
-| `status`  | string array | Vault status -- optional, see [vault statuses](#vault-statuses) for possible values             |
-| `txids`   | string array | Vault IDs -- optional, filter the list with the given vault IDs                                 |
+| Parameter   | Type         | Description                                                                                     |
+| ----------- | ------------ | ----------------------------------------------------------------------------------------------- |
+| `status`    | string array | Vault status -- optional, see [vault statuses](#vault-statuses) for possible values             |
```
Comment (edouardparis) [inline doc/API.md:101]: nit: order fields

File: src/daemon/database/interface.rs (hunk 1 above)
Comment (darosior) [reply src/daemon/database/interface.rs:195]: You may :p 
Addressed!

File: doc/API.md (hunk 2 above)
Comment (darosior) [reply doc/API.md:101]: Addressed..

**Comment by darosior** [comment]
Ok, i think it's now gtg

**Review by edouardparis — APPROVED** [review]
ACK 4c7c1e637ec7c579491bb69ea4a4ab5d3c138eaa

**Pull request merged**